### PR TITLE
DM-20558 Rescale the coadd variances for difference imaging templates

### PR DESCRIPTION
**Pull request by kgabor**

### CODE REVIEW

File: python/lsst/pipe/tasks/imageDifference.py [hunk 1]
```diff
@@ -167,6 +175,7 @@ class ImageDifferenceConfig(pexConfig.Config):
         target=RegisterTask,
         doc="Task to enable image-to-image image registration (warping)",
     )
+
```
Comment (yalsayyad) [inline python/lsst/pipe/tasks/imageDifference.py]: Accidental whitespace added between configs

File: python/lsst/pipe/tasks/imageDifference.py [hunk 2]
```diff
@@ -191,6 +200,7 @@ def setDefaults(self):
         self.subtract['al'].kernel.active.spatialKernelOrder = 1
         self.subtract['al'].kernel.active.spatialBgOrder = 2
         self.doPreConvolve = False
+        self.doScaleTemplateVariance = False
```
Comment (yalsayyad) [inline python/lsst/pipe/tasks/imageDifference.py]: Remove this. It's redundant, with the original default you specified above. 

In fact, every default in this set defaults method that is NOT a subtask config should be moved above.

File: python/lsst/pipe/tasks/scaleVariance.py [hunk 3]
```diff
@@ -26,6 +26,8 @@
 from lsst.pipe.base import Task
 from lsst.meas.algorithms import SubtractBackgroundTask
 
+__all__ = ["ScaleVarianceConfig", "ScaleVarianceTask"]
+
```
Comment (yalsayyad) [inline python/lsst/pipe/tasks/scaleVariance.py]: This is part of the "Add docs" commit, right.

Comment (kgabor) [reply python/lsst/pipe/tasks/scaleVariance.py]: Automodapi may work without this, but I think it's better to have it explicitly written for it.

File: python/lsst/pipe/tasks/imageDifference.py (hunk 2 above)
Comment (kgabor) [reply python/lsst/pipe/tasks/imageDifference.py]: These actually override the default values, I can't see why.

**Pull request merged**